### PR TITLE
OVAL service templates should check if service is running/not running

### LINKED ISSUE #2064: DISA: SSG does not rate-limit via firewalld
> RHEL-07-040510
The operating system must protect against or limit the effects of Denial of Service (DoS) attacks by validating the operating system is implementing rate-limiting measures on impacted network interfaces.
----
> "DoS is a condition when a resource is not available for legitimate users. When this occurs, the organization either cannot accomplish its mission or must operate at degraded capacity. This requirement addresses the configuration of the operating system to mitigate the impact of DoS attacks that have occurred or are ongoing on system availability. For each system, known and potential DoS attacks must be identified and solutions for each type implemented. A variety of technologies exist to limit or, in some cases, eliminate the effects of DoS attacks (e.g., limiting processes or establishing memory partitions). Employing increased capacity and bandwidth, combined with service redundancy, may reduce the susceptibility to some DoS attacks."
----
> "Verify the operating system protects against or limits the effects of DoS attacks by ensuring the operating system is implementing rate-limiting measures on impacted network interfaces.
Check the firewall configuration with the following command:
Note: The command is to query rules for the public zone.
 firewall-cmd --direct --get-rule ipv4 filter IN_public_allow rule ipv4 filter IN_public_allow 0 -p tcp -m limit --limit 25/minute --limit-burst 100  -j ACCEPT
If a rule with both the limit and limit-burst arguments parameters does not exist, this is a finding."
----
> Create a direct firewall rule to protect against DoS attacks with the following command:
Note: The command is to add a rule to the public zone.
firewall-cmd --direct --add-rule ipv4 filter IN_public_allow 0 -p tcp -m limit --limit 25/minute --limit-burst 100  -j ACCEPT"
-----
> CCI-002385
The information system protects against or limits the effects of organization-defined types of denial of service attacks by employing organization-defined security safeguards.
NIST SP 800-53 Revision 4 :: SC-5

### LINKED ISSUE #2065: DISA: service_firewalld_enabled does not check if service is enabled
DISA:
> I found the reference file in https://github.com/OpenSCAP/scap-security-guide/blob/d0af7b820664fe193195eebe498840b654bdf509/RHEL/7/input/oval/oval_5.11/service_firewalld_enabled.xml but it's looking for dependencies, not that the service is active

### PR DESCRIPTION
**Pull request by redhatrises**
#### Description:

- OVAL service templates should check if service is running/not running

#### Rationale:

- Current OVAL 5.11 service templates only check if a service is enabled/disabled but not if it is running/not running.

- Fixes #2065

### CODE REVIEW

File: rhel7/templates/oval_5.11_templates/template_OVAL_service_enabled [hunk 1]
```diff
@@ -33,12 +34,22 @@
...
-
   <linux:systemdunitdependency_object id="object_multi_user_target_for_%SERVICENAME%_socket_enabled" comment="list of dependencies of multi-user.target" version="1">
     <linux:unit>multi-user.target</linux:unit>
   </linux:systemdunitdependency_object>
-
   <linux:systemdunitdependency_state id="state_systemd_%SERVICENAME%_socket_on" comment="%SERVICENAME% listed at least once in the dependencies" version="1">
     <linux:dependency entity_check="at least one">%SERVICENAME%.socket</linux:dependency>
   </linux:systemdunitdependency_state>
+
+  <linux:systemdunitproperty_test id="test_service_running_%SERVICENAME%" check="all" check_existence="at_least_one_exists" comment="Test that the %SERVICENAME% service is running" version="1">
```
Comment (yuumasato) [inline rhel7/templates/oval_5.11_templates/template_OVAL_service_enabled]: All `test_service_running_` should have `check="at least one"`. It is enough that either service or socket is in active state.

File: rhosp7/templates/oval_5.11_templates/template_OVAL_service_disabled [hunk 2]
```diff
@@ -33,12 +32,22 @@
...
-
   <linux:systemdunitdependency_object id="object_multi_user_target_for_%SERVICENAME%_socket_disabled" comment="list of dependencies of multi-user.target" version="1">
     <linux:unit>multi-user.target</linux:unit>
   </linux:systemdunitdependency_object>
-
   <linux:systemdunitdependency_state id="state_systemd_%SERVICENAME%_socket_off" comment="%SERVICENAME% socket is not listed in the dependencies" version="1">
     <linux:dependency entity_check="none satisfy">%SERVICENAME%.socket</linux:dependency>
   </linux:systemdunitdependency_state>
+
+  <linux:systemdunitproperty_test id="test_service_not_running_%SERVICENAME%" check="all" check_existence="at_least_one_exists" comment="Test that the %SERVICENAME% service is not running" version="1">
```
Comment (yuumasato) [inline rhosp7/templates/oval_5.11_templates/template_OVAL_service_disabled]: All `test_service_not_running_` should have `check_existence="any_exist", as it should be ok to not find the unit.

**Comment by yuumasato** [comment]
@redhatrises Is the reference to the issue in "Fixes" list correct? I don't understand the relationship.

**Comment by redhatrises** [comment]
> @redhatrises Is the reference to the issue in "Fixes" list correct? I don't understand the relationship.

Thanks @yuumasato. The reference to the issue was a typo.

**Comment by yuumasato** [comment]
Ack.

**Pull request merged**